File: ts/src/parkingBoy/seniorParkingBoy.ts
/** @format */

import {MostAvailableParkingBoyService} from './mostAvailableParkingBoyService'
import {ParkingLot} from '../parkingLot/parkingLot'
import {BaseParkingBoy} from './baseParkingBoy'

export class SeniorParkingBoy extends BaseParkingBoy {
    public constructor(parkingLots: ParkingLot[]) {
        super(parkingLots, new MostAvailableParkingBoyService())
    }
}

File: cs/Parking/Boy/SeniorParkingBoy.cs
using System.Collections.Generic;

namespace Parking
{
    public class SeniorParkingBoy : BaseParkingBoy
    {
        public SeniorParkingBoy(IReadOnlyList<ParkingLot> parkingLots) : base(parkingLots, new MostAvailableParkingBoyService()) { }
    }
}
